server/application-server/src/main/java/de/tum/cit/aet/helios/environment/EnvironmentService.java

```diff
@@ -631,4 +642,31 @@ public Optional<EnvironmentReviewersDto> getEnvironmentReviewers(Long environmen
               }
             });
   }
+
+  /** Synchronizes the environments of the current repository with the GitHub repository. */
+  public void syncRepositoryEnvironments() {
+    Long repoId = RepositoryContext.getRepositoryId();
+    GitRepository repo = gitRepoRepository.findById(repoId).orElseThrow();
+    GHRepository ghRepository;
+    try {
+      ghRepository = this.gitHubService.getRepository(repo.getNameWithOwner());
+    } catch (IOException e) {
+      log.error("Failed to get GitHub repository: {}", e.getMessage());
```

Comment:
issue: Please add a throw statement here as currently it returns fetched successful when there is no repository found. You can use the EntityNotFound error here.